gobblin-data-management/src/main/java/gobblin/data/management/retention/action/AccessControlAction.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.data.management.retention.action;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsPermission;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.typesafe.config.Config;
+
+import gobblin.data.management.policy.VersionSelectionPolicy;
+import gobblin.data.management.retention.dataset.ConfigurableCleanableDataset;
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.data.management.version.FileSystemDatasetVersion;
+import gobblin.util.ConfigUtils;
+import gobblin.util.reflection.GobblinConstructorUtils;
+
+
+/**
+ * A {@link RetentionAction} that is used to change the permissions/owner/group of a {@link FileSystemDatasetVersion}
+ */
+@Slf4j
+public class AccessControlAction extends RetentionAction {
+
+  /**
+   * Optional - The permission mode to set on selected versions. E.g 750
```

Comment:
The javadoc for FsPermission say that this can be either and octal number or symbolic. I think it is worth mentioning here.